### ISSUE TITLE
Support Zcash NU5

### ISSUE THREAD

**Issue by andrewkozlik** [trezor/trezor-suite#5344]
Zcash will soon [1] deploy the NU5 upgrade to mainnet. Currently NU5 is running on Zcash testnet. Suite only supports v4 transactions. As a result of the upgrade Suite users will not be able to spend UTXOs created in v5 transactions. However, they will still be able to spend UTXOs created in v4 transactions and earlier-version transactions.

In order to spend a UTXO created in a v5 transaction, Suite needs to spend it in a v5 transaction, which can be done as follows:
- The `version` field in `SignTx` needs to be set to 5.
- The `branchId` needs to be set to 0xc2d6d0b4, see https://zips.z.cash/zip-0252.
- The `versionGroupId` needs to be set to 0x26a7270a, I think (no source available).
- When signing v5 transactions the client does not need to provide the previous transaction(s).

The accounts and addresses remain the same as for version 4. The serialized transaction structure is different, but that is taken care of by Trezor firmware. Suite should not need to make any updates in this regard, just broadcast the transaction that Trezor spits out.

There are two implementation options for signing Zcash transactions:
1. Once this is implemented Suite will only create v5 transactions.
2. Suite will create a v5 transaction only if at least one transaction input comes from a v5 transaction. Otherwise it will create a v4 transaction.

The advantage of Option 1 is that no prev_tx streaming is needed. The advantage of Option 2 is that we only create v5 UTXOs if it's necessary to spend our selected UTXOs. This way if the wallet used by the recipient of the payment does not support v5 transactions yet, they might still be able to receive the payment, i.e. see that the payment was received and be able to spend it.

We can implement Option 2 for a transition period and switch to Option 1 later.

On Trezor NU5 support will be available starting with the upcoming releases of T1 firmware v1.11.1 and TT firmware v2.5.1. If Suite intends to sign a v5 transaction, it will need to ensure that the user has this firmware version or a later version.

This issue might be related to https://github.com/trezor/trezor-suite/issues/3749 (maybe `branchId` and `versionGroupId` can be retrieved from blockbook?). But implementing it is certainly not a prerequisite for this issue.

[1] Zcash schedule: https://z.cash/support/schedule/
[2] Firmware issue: https://github.com/trezor/trezor-firmware/issues/2031
[3] TT firmware PR: https://github.com/trezor/trezor-firmware/pull/2166
[4] T1 firmware PR: https://github.com/trezor/trezor-firmware/pull/2244

**Comment by prusnak** [comment]
Mainnet NU5 upgrade activates on 31st of May - see https://z.cash/upgrade/nu5/ - this should get higher priority and an assignee.

**Comment by tsusanka** [comment]
As discussed: we will wait with this one till mid-June to see what the demand is. Then we'll implement right away or a bit later. @hynek-jina will instruct support to monitor the situation.

**Comment by deepcandles** [comment]
Hi,

Zcash community is very much interested in v5 transactions, unified addresses & using Orchard shielded pool through Trezor. In fact, Zcash community has been waiting for years. I would recommend not waiting for mid-june.

You can clearly see the community excitement: 
https://twitter.com/deepcandles/status/1509435834231001089
https://twitter.com/deepcandles/status/1526768137714028546

Plus I’m sure you know, Zcash community is funding this work only because it’s interested in this for a long long time. 

cc @prusnak 

Thank you!

**Comment by nighthawk24** [comment]
Agree with @deepcandles 💯

The Zcash community funded this integration and is very eager to use it on their Trezor Model-Ts. Several Zcashers have purchased their first Trezor for this very integration.

Please monitor the NU5 upgrade going live on May 31st ~16:00 UTC

**Comment by Z2z123** [comment]
Definitely agree, can’t wait for this integration. Will definitely be swapping from Ledger to a few Trezor T’s once integration date is confirmed and I know there are many others in the same boat

**Comment by prusnak** [comment]
We know there is a demand for this feature, no need to write duplicate comments about it. Reading them takes developers' time which could have been spent elsewhere.

This particular issue is about v5 transaction support (which has been already implemented in Trezor Firmware) and this is going to be activated in Trezor Suite soon.

Shielded transactions are being tracked in another (firmware) issue - https://github.com/trezor/trezor-firmware/pull/1847

TL;DR: please keep the discussion related to technical perspective of things.

**Comment by prusnak** [comment]
As Andrew pointed out in the OP (Option 1), v5 TX support could be trivially achieved by changing these constants at the right moment (after NU5 is activated):

``` patch
diff --git a/packages/suite/src/constants/wallet/sendForm.ts b/packages/suite/src/constants/wallet/sendForm.ts
index 81f7c8a3c..25ab5e94c 100644
--- a/packages/suite/src/constants/wallet/sendForm.ts
+++ b/packages/suite/src/constants/wallet/sendForm.ts
@@ -10,9 +10,9 @@ export const ETH_DEFAULT_GAS_LIMIT = '21000';
 
 export const ZEC_SIGN_ENHANCEMENT = {
     overwintered: true,
-    version: 4,
-    versionGroupId: 0x892f2085,
-    branchId: 0xe9ff75a6,
+    version: 5,
+    versionGroupId: 0x26a7270a,
+    branchId: 0xc2d6d0b4,
 };
 
 export const ERC20_TRANSFER = 'a9059cbb'; // 4 bytes function signature of solidity erc20 `transfer(address,uint256)`
```

However, this approach has a disadvantage that users might send funds to a wallet that has not been NU5-enabled yet and they will not be able to spend them from there. I'll ask what's the state of other wallet implementations in the ZEC ecosystem and report back.

Option 2 is more convoluted.

**Comment by prusnak** [comment]
I implemented Option 1 in https://github.com/trezor/trezor-suite/pull/5453

If the support table at the bottom of https://z.cash/upgrade/nu5/ should be trusted, there should not be an issue to support v5 only straight away.

**Comment by deepcandles** [comment]
Thanks @prusnak for clear explanation. I understand it better.

**Comment by luckytokidoki** [comment]
fwiw v5 transactions fix the txid malleability problem. cheers!

**Comment by andrewkozlik** [comment]
Update: NU5 has been activated on Mainnet https://electriccoin.co/blog/nu5-activates-on-mainnet-eliminating-trusted-setup-and-launching-a-new-era-for-zcash/.

**Comment by bosomt** [comment]
**QA OK**

https://zec3.trezor.io/tx/b17073805c4552d5f3db6d60e044088fe1ac015971964ee4ebf3a4124978104c

**Info:**
 - Suite version: desktop 22.6.2 (f45c75705f153ec0dc44cb72fec2e5ec4a031b7c)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/22.6.2 Chrome/94.0.4606.81 Electron/15.3.0 Safari/537.36
 - OS: MacIntel
 - Screen: 1680x1050
 - Device: model T 2.5.1 regular (revision 85a26d2c9593bcdf858c2d718d79951ca927a0c3)
 - Transport: bridge 2.0.31
